src/node-file-trace.ts

```diff
@@ -270,13 +284,20 @@ export class Job {
   }
 
   async getPjsonBoundary (path: string) {
+    const initialPath = path
+    const cacheItem = this.pjsonBoundaryCache.get(path)
+    if (typeof cacheItem !== 'undefined') return cacheItem
```

Comment:
This cache may be redundant since the only FS operations this method depends on is `isFile` which uses the stat cache, which uses `null` for empty entries so does fully cache both existing and non existing package.json files already. The loop saving should be negligible since file depths usually aren't that large.

So for simplicity it may well be worth removing this `pjsonBoundaryCache` and just relying on the stat cache - there shouldn't be any difference in performance.

Reply:
Yeah makes sense, double checked and it looks like it doesn't provide any difference in performance so removed it to reduce the additional complexity.